File: core/src/main/java/net/siegerpg/siege/core/drops/materials/CHAIN.java
package net.siegerpg.siege.core.drops.materials;

import net.siegerpg.siege.core.drops.BlockDropTable;
import net.siegerpg.siege.core.drops.Reward;
import net.siegerpg.siege.core.items.implemented.misc.materials.drops.blocks.Chain;
import org.bukkit.Material;

public class CHAIN extends BlockDropTable {

	public CHAIN() {

		super(40, Material.CHAIN, 0, 0, 0, 0, new Reward[] {
				new Reward(new Chain().getUpdatedItem(false), 100.0)
		});
	}

}

File: core/src/main/kotlin/net/siegerpg/siege/core/items/implemented/misc/materials/drops/blocks/Chain.kt
package net.siegerpg.siege.core.items.implemented.misc.materials.drops.blocks

import net.siegerpg.siege.core.items.CustomItemUtils
import net.siegerpg.siege.core.items.enums.Rarity
import net.siegerpg.siege.core.items.types.misc.CustomMaterial
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class Chain() : CustomMaterial(
		name = "Chain",
		customModelData = 320007,
		description = listOf("A strong metal rope"),
		levelRequirement = 0,
		material = Material.FLINT,
		sellCost = 5
                              ) {

	constructor(quality : Int) : this() {
		this.quality = quality
		this.rarity = Rarity.getFromInt(quality)
		this.serialize()
	}

	constructor(item : ItemStack) : this() {
		this.item = item
		deserialize()
	}

	companion object {

		fun tier(tier : Int) : Chain {
			val newItem = Chain(0)
			newItem.tier = tier
			return newItem
		}
	}

}